### PR TITLE
Aq 74 create player view

### PR DESCRIPTION
**Pull request by MWasile**
# Description

added view and serializer for player model, which enable creating new players with a nickname
## Type of change

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

6 unit tests 

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] The related memme has been added

img

### CODE REVIEW

**Review by martynawitkowska — CHANGES_REQUESTED** [review]

**Comment by SlawCzech** [comment]
Everything works fine for me. Create-player works, tests pass. One side-comment though - in models I would change 'is_bot' field 'verbose name' to "Bot player", because showing only "Player type" in admin is very unclear.

**Review by SlawCzech — APPROVED** [review]
Works fine for me.

File: quiz/proxy/default.conf [hunk 1]
```diff
@@ -22,6 +22,13 @@ server {
         alias /app/staticfiles/;
     }
 
+    location /api/v1/ {
```
[the remark this answers is not in the document]
Comment (ToJestKrzysio) [reply quiz/proxy/default.conf:25]: seems fine for now

File: quiz/backend/players/tests/test_serializers.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
+from datetime import datetime
```
Comment (martynawitkowska) [inline quiz/backend/players/tests/test_serializers.py:1]: unused import

File: quiz/backend/players/tests/test_views.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import json
```
Comment (martynawitkowska) [inline quiz/backend/players/tests/test_views.py:1]: unused import

**Review by martynawitkowska — APPROVED** [review]
Approving, but remove unused imports :D

File: quiz/backend/players/tests/test_views.py [hunk 4]
```diff
@@ -0,0 +1,15 @@
+from players.views import PlayerCreateAPIView
+
+
+def test_player_view_return_201_status_code_and_correct_data(player_api):
+    response = player_api()
+
+    assert response.status_code == 201
+    assert response.data["nick"] == "test_nick"
```
Comment (ToJestKrzysio) [inline quiz/backend/players/tests/test_views.py:8]: "test_nick" could be generated by fixture and used by test and fixture generating data.

**Review by ToJestKrzysio — COMMENTED** [review]
Just notes

**Pull request merged**